src/sdk/service/network-validation-service/network-validation-service.ts

```diff
@@ -0,0 +1,475 @@
+import { Range, NetworkParams } from './network-validation-interface';
+
+/**
+ * Network Validation Service
+ */
+export class NetworkValidationService {
+  private static instance: NetworkValidationService;
+
+  /**
+   * @return {NetworkValidationService} instance of NetworkValidationService class.
+   */
+  static getInstance(): NetworkValidationService {
+    if (!NetworkValidationService.instance) {
+      NetworkValidationService.instance = new NetworkValidationService();
+    }
+    return NetworkValidationService.instance;
+  }
+
+  /**
+   * Gets the integer representation of an IP address.
+   * @param {string} ip: the ip address.
+   * @return {number} The integer representation.
+   */
+  getIpAsInteger(ipAddress: string): number | null {
```

Comment:
all method names should include IPv4, if relevant because we'll be adding similar methods for IPv6 in the near future and it's better to be explicit